src/site_upload/study_period/study_period.py

```diff
@@ -11,12 +11,23 @@
 
 def update_study_period(s3_client, s3_bucket, site, study, data_package, version):
     """gets earliest/latest date from study metadata files"""
-    path = awswrangler_functions.get_s3_study_meta_list(
+    paths = awswrangler_functions.get_s3_study_meta_list(
         s3_bucket, study, data_package, site, version
     )
-    if len(path) != 1:
-        raise KeyError("Unique date path not found")
-    df = awswrangler.s3.read_parquet(path[0])
+    if len(paths) > 1:
+        latest_key = None
+        latest_date = datetime(1000, 1, 1, tzinfo=UTC)
+        for path in paths:
+            key = functions.get_s3_key_from_path(path)
+            res = s3_client.head_object(Bucket=s3_bucket, Key=key)
+            if res["LastModified"] > latest_date:
+                latest_key = key
+                latest_date = res["LastModified"]
+        for path in paths:
+            if latest_key not in path:
```

Comment:
I always get so nervous when I saw a bare `in` check like this - worried about false positives. Would you expect multiple paths to match normally? Can we use `latest_path` instead of `latest_key` and then just do an equality check? Or if we like keeping the key only, maybe use `endswith` instead of `in`?

Reply:
so since these are file paths inside of a bucket, and we're not having users upload directly, they should be globally unique and unlikely to be substrings

i can use path instead, but key is the thing that the boto tools expect, so that's what we want to use later.

Reply:
Sure, but you don't use `latest_key` later in this case. But I get ya.

Reply:
yeah, i ended up pulling it out anyway as part of moving to max(function) as you suggested